### PR TITLE
Utilise LocalProjectBuilder for LP initalisation

### LINKED ISSUE #1955: Use LocalProbjectBuilder instead of old LocalProject constructor
Follow-up to https://github.com/packit/packit/pull/1558 

The old usages should be removed, tests updated. Inspiration can be taken from adjustments in Packit CLI in the linked PR

Also consider if there are not unnecessary arguments/attributes in the API in the new class, e.g. we were debating whether `remote` and `pr_id` and similar still make sense.

EDIT [Jirka]: FYI, hardly code should be updated in packit/hardly#92

### PR DESCRIPTION
**Pull request by lbarcziova**
With this change, for copr builds also try to not clone the repo.

Fixes #1955
Related to #2411

For now no release notes, I would like to see whether this will work on staging as intended.

RELEASE NOTES BEGIN

N/A

RELEASE NOTES END

### CODE REVIEW

**Comment by lbarcziova** [comment]
recheck

**Comment by lbarcziova** [comment]
recheck

**Comment by danpawlik** [comment]
recheck

**Comment by danpawlik** [comment]
recheck

**Comment by danpawlik** [comment]
recheck

File: packit_service/worker/helpers/job_helper.py [hunk 1]
```diff
@@ -91,7 +94,22 @@ def local_project(self) -> LocalProject:
...
                 merge_pr=self.package_config.merge_pr_in_ci,
+                git_url=CALCULATE,
+                repo_name=CALCULATE,
+                full_name=CALCULATE,
+                namespace=CALCULATE,
+                git_repo=(
+                    CALCULATE
+                    if self.require_git_repo_in_local_project
+                    else NOT_TO_CALCULATE
+                ),
```
Comment (lachmanfrantisek) [inline packit_service/worker/helpers/job_helper.py:112]: Nicely explicit what is going on here... Finally getting some fruits from work on all of this...;)

**Review by lachmanfrantisek — APPROVED** [review]
Looks good to me! Even the tests did not require massive changes...

+1 for trying this on stage

**Pull request merged**